src/beatmap.c

```diff
@@ -185,6 +185,28 @@ static void parse_one_hit(char *line, struct oshu_hit **hit)
 }
 
 /**
+ * Parse #oshu_hit::combo and #oshu_hit::combo_seq of a single #oshu_hit
+ */
+static void parse_hit_combo(struct oshu_hit *previous, struct oshu_hit *hit)
+{
+	int new_combo = hit->type & OSHU_HIT_NEW_COMBO;
+	int skip_combo = (hit->type & OSHU_HIT_COMBO_MASK) >> 4;
```

Comment:
4 is a magic number here. Could you add `OSHU_HIT_COMBO_OFFSET=4` in the enum and use that instead?

By the way, it could be defined inside the new_combo block to follow the proximity rule of grouping related things closely together.